### PR TITLE
[execute_process] Don't strip embedded semicolons

### LINKED ISSUE #12922: show-stopper: vcpkg screwing up cmake by removing semicolons
`execute_process(COMMAND "C:/msys2/msys64/usr/bin/echo.exe" "hello; world; line 3" OUTPUT_VARIABLE MY_OUT)`
`message(STATUS "result ${MY_OUT}")`
should result in:
`"-- result hello; world; line 3"` (which is does when just running it in a minimalistic cmake project not using vcpkg)
but actually results in:
`"-- result hello  world  line 3"`
when this code is placed in a vcpkg cmake port file. No matter what the intention behind this behaviour is, this sooner or later becomes a show-stopper, for example, when trying to find python using
`find_package(Python3)`
or using the old cmake API
`find_package(PythonInterp)`
because cmake internally tries to run
`python -c "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"`
to find the correct version, which fails because python stops on a syntax error after "import sys" because of the erased semicolon.

This occurres at least on Windows 10 (where I found it).

### PR DESCRIPTION
**Pull request by jgehw**
The existing implementation totally screwed up commands if the command's arguments contained semicolons (this is the case, e.g., in the FindPython modules of the cmake distribution).

Fixes #12922.

### CODE REVIEW

**Comment by jgehw** [comment]
Added support for execute_process with multiple COMMANDs as there actually are use cases in both vcpkg itself and some ports.

File: scripts/cmake/execute_process.cmake [hunk 1]
```diff
@@ -13,8 +13,75 @@ if (NOT DEFINED OVERRIDEN_EXECUTE_PROCESS)
       message(FATAL_ERROR "This command cannot be executed in Download Mode.\nHalting portfile execution.\n")
     endmacro()
   else()
```
Comment (BillyONeal) [inline scripts/cmake/execute_process.cmake:15]: In this `else` case could we just do nothing instead?

**Comment by jgehw** [comment]
Why did you close this PR? #12977 is no fix for this case. We either need this fix or [ras0219@b76640e].

**Comment by BillyONeal** [comment]
@jgehw I didn't actively try to close this, GitHub does that itself if a PR is merged that mentions another issue or PR with a hot phrase like "fixes 1234" in the discussion / history.

**Comment by BillyONeal** [comment]
(Specifically that PR had "fixed #12926" in the description which causes GitHub to link them together and close one when the other is merged)

**Comment by jgehw** [comment]
@BillyONeal Ahh, I see, I'll pay attention to avoid such phrasing next time.

**Comment by jgehw** [comment]
Found and fixed a bug in regex matching, which occurred in CI testing.

**Comment by JackBoosY** [comment]
`execute_process` is often used in vcpkg, we must carefully modify this function.

**Comment by jgehw** [comment]
> `execute_process` is often used in vcpkg, we must carefully modify this function.

- I implemented it with care, always knowing what I'm doing.
- Nevertheless I made a mistake (forgot to escape regex special chars before matching, see fix in last commit). This mistake was detected in CI testing.
- Beside CI testing, I suggest to have three independent code reviewers, not just one.

Do think this should suffice? If not, do you have any more ideas how to add additional layers of care?

**Comment by BillyONeal** [comment]
I would like to see Robert's version that doesn't keep us needing to constantly sync with the upstream definition of execute_process.

**Comment by jgehw** [comment]
ok, I replaced my solution with Robert's implementation

**Comment by BillyONeal** [comment]
@ras0219 OK we tried your workaround but it causes stuff to break.

Maybe we should change all the `_execute_process` calls to `vcpkg_execute_in_download_mode` or similar?

**Comment by jgehw** [comment]
> Maybe we should change all the `_execute_process` calls to `vcpkg_execute_in_download_mode` or similar?

@BillyONeal Did this as you suggested. I'm curious for the CI results ;-)

**Comment by jgehw** [comment]
I don't see a link between last nights build errors and this PR's code changes. Some errors sound transient, e.g.
`-- The default mirror is in Disaster Recovery mode, trying other mirrors...`; others such as unresolved symbols during linking sound unrelated -- merging master branch to ensure we're not missing any fixes here.

**Comment by jgehw** [comment]
The errors are gone now except for the selene port, but I saw this also failing in other branches, thus most likely an issue with selene.

**Comment by jgehw** [comment]
arm_uwp and x64_uwp CI failing on json5-parser port downloading https://bitbucket.com/wlandry/json5_parser/get/1.0.0.tar.gz. When viewing in a browser, I get "Repository wlandry/json5_parser not found". So, not related to this PR.

**Comment by BillyONeal** [comment]
Yeah, upstream is gone.

**Review by strega-nil — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by jgehw** [comment]
upstream is gone again... please re-run once upstream is fixed

**Pull request merged**